be/src/exec/pipeline/scan/morsel.h

```diff
@@ -137,14 +148,37 @@ class ScanMorsel : public Morsel {
         return std::tuple<int64_t, int64_t>{_owner_id, _version};
     }
 
+    void set_split_context(ScanSplitContextPtr&& split_context) {
+        if (split_context == nullptr) return;
+        _split_context = std::move(split_context);
+        _is_last_split = _split_context->is_last_split;
+    }
+    ScanSplitContext* get_split_context() {
+        if (_split_context != nullptr) {
+            return _split_context.get();
+        }
+        return nullptr;
```

Comment:
`std::unique_ptr<>(nullptr).get()` is also safe.
```suggestion
        return _split_context.get();
```